packages/nextcloud/packages/nextcloud_test_presets/lib/src/models/app_release.dart

```diff
@@ -16,14 +16,15 @@ class AppRelease {
   final Version minimumServerVersion;
   final Version maximumServerVersion;
 
-  Version? findLatestServerVersion(List<Version> serverVersions) {
-    final compatibleReleases = serverVersions
+  ServerRelease? findLatestServerRelease(List<ServerRelease> serverReleases) {
+    final compatibleReleases = serverReleases
         .where(
-          (serverVersion) =>
-              serverVersion >= minimumServerVersion && serverVersion < maximumServerVersion.incrementMajor(),
+          (serverRelease) =>
+              serverRelease.version >= minimumServerVersion &&
+              serverRelease.version < maximumServerVersion.incrementMajor(),
```

Comment:
With the server release being a separate class this should be a method in there. Maybe something like `bool isInRange(ServerRelease other)`.